Library/MapleBacon/MapleBacon/Resizer.swift

```diff
@@ -6,176 +6,251 @@ import UIKit
-            newX = offset.width
-            newY = 0
-        case .BottomLeft:
-            newX = 0
-            newY = offset.height
-        case .BottomRight:
-            newX = offset.width
-            newY = offset.height
+        
+        var resizeData = ResizeCalculationEntity(currentSize: image.size, intendedSize: size)
+        resizeData.applyContentMode(contentMode: contentMode)
+        
+        let newSize   = resizeData.newSize.scaled()
+        let newBounds = resizeData.resizeBounds.scaled()
+        
+        let action = {
+            self.resize(toSize: newSize, toBounds: newBounds, interpolationQuality: quality, async: async, completion: completion)
         }
 
-        let scaleDependentNewSize = CGSize(width: newSize.width * deviceScale, height: newSize.height * deviceScale)
-        let newBounds = CGRect(x: newX * deviceScale, y: newY * deviceScale, width: size.width * deviceScale,
-                height: size.height * deviceScale)
-
         if async {
-            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) {
-                self.resizeImageFromImage(image, toSize: scaleDependentNewSize, newBounds: newBounds,
-                        interpolationQuality: quality, async: async, completion: completion)
-            }
-        } else {
-            self.resizeImageFromImage(image, toSize: scaleDependentNewSize, newBounds: newBounds,
-                    interpolationQuality: quality, async: async, completion: completion)
+            
+            DispatchQueue.global().async(execute: action)
+            return
         }
+        
+        action()
     }
-
-    private class func resizeImageFromImage(image: UIImage, toSize size: CGSize, newBounds bounds: CGRect,
-                                            interpolationQuality quality: CGInterpolationQuality, async: Bool,
-                                            completion: ResizerCompletion) {
-        var imageToReturn = image
-        if let resizedImage = imageFromImage(image, toSize: size, interpolationQuality: quality),
-        let croppedImage = croppedImageFromImage(resizedImage, toBounds: bounds) {
+    
+    /**
+     Prequisite reszing method
+     */
+    private func resize(toSize size: CGSize, toBounds bounds: CGRect, interpolationQuality quality: CGInterpolationQuality, async: Bool, completion: ResizerCompletion) {
+        
+        var imageToReturn = self.image
+        
+        let drawTransposed: Bool
+        switch image.imageOrientation {
+        case .left, .leftMirrored, .right, .rightMirrored:
+            drawTransposed = true
+        default:
+            drawTransposed = false
+        }
+        
+        if
+            let resizedImage = resize(toSize: size, usingTransform: transformForOrientation(toSize: size), drawTransposed: drawTransposed, interpolationQuality: quality),
+            let croppedImage = crop(image: resizedImage, toBounds: bounds)
+        {
```

Comment:
not a fan of the formatting, `if let` on one line and opening braces on the same line as the last statement. And multiple `let`s as above somewhere. 

```swift
if let resizedImage = …, croppedImage = … {}
```